### PR TITLE
Fix ModuleNotFoundError: No module named 'skimage' when importing torcheval

### LINKED ISSUE #204: Cannot `from torcheval.metrics import *` when skimage is not installed
### 🐛 Describe the bug

When importing all metrics, the import will fail if `skimage` is not installed. This also happens when importing an unrelated metric such as `F1Score`.

Reproducer:
```
import torch
import torcheval
from torcheval.metrics import F1Score
# from torcheval.metrics import *
```
will raise the following error on main:
```
/home/xy/torcheval/lib/python3.11/site-packages/torch/_subclasses/functional_tensor.py:258: UserWarning: Failed to initialize NumPy: No module named 'numpy' (Triggered internally at ../torch/csrc/utils/tensor_numpy.cpp:84.)
  cpu = _conversion_method_template(device=torch.device("cpu"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/xy/torcheval/lib/python3.11/site-packages/torcheval/metrics/__init__.py", line 55, in <module>
    from torcheval.metrics.image import (
  File "/home/xy/torcheval/lib/python3.11/site-packages/torcheval/metrics/image/__init__.py", line 11, in <module>
    from torcheval.metrics.image.ssim import StructuralSimilarity
  File "/home/xy/torcheval/lib/python3.11/site-packages/torcheval/metrics/image/ssim.py", line 16, in <module>
    from skimage.metrics import structural_similarity
ModuleNotFoundError: No module named 'skimage'
```

Happens on branch main, commit 7b999093812983c4a999c44f1aacb5d010143e3f

### Versions

Collecting environment information...
PyTorch version: 2.4.0+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 23.10 (x86_64)
GCC version: (Ubuntu 13.2.0-4ubuntu3) 13.2.0
Clang version: Could not collect
CMake version: version 3.27.4
Libc version: glibc-2.38

Python version: 3.11.6 (main, Apr 10 2024, 17:26:07) [GCC 13.2.0] (64-bit runtime)
Python platform: Linux-6.5.0-44-generic-x86_64-with-glibc2.38
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3080 Ti Laptop GPU
Nvidia driver version: 545.23.08
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      46 bits physical, 48 bits virtual
Byte Order:                         Little Endian
CPU(s):                             14
On-line CPU(s) list:                0-13
Vendor ID:                          GenuineIntel
Model name:                         12th Gen Intel(R) Core(TM) i7-12800H
CPU family:                         6
Model:                              154
Thread(s) per core:                 1
Core(s) per socket:                 14
Socket(s):                          1
Stepping:                           3
CPU(s) scaling MHz:                 50%
CPU max MHz:                        4800,0000
CPU min MHz:                        400,0000
BogoMIPS:                           5606,40
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l2 cdp_l2 ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdt_a rdseed adx smap clflushopt clwb intel_pt sha_ni xsaveopt xsavec xgetbv1 xsaves split_lock_detect avx_vnni dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req hfi vnmi umip pku ospke waitpkg gfni vaes vpclmulqdq tme rdpid movdiri movdir64b fsrm md_clear serialize pconfig arch_lbr ibt flush_l1d arch_capabilities
Virtualization:                     VT-x
L1d cache:                          544 KiB (14 instances)
L1i cache:                          704 KiB (14 instances)
L2 cache:                           11,5 MiB (8 instances)
L3 cache:                           24 MiB (1 instance)
NUMA node(s):                       1
NUMA node0 CPU(s):                  0-13
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] torch==2.4.0
[pip3] torcheval==0.0.7
[pip3] triton==3.0.0
[conda] Could not collect

### PR DESCRIPTION
**Pull request by fleonce**
Summary:
Prevent ModuleNotFoundError when skimage is not installed

Test plan:
None

Fixes #204

### CODE REVIEW

**Comment by bobakfb** [comment]
Thanks for this fix! Testing!

File: torcheval/metrics/image/ssim.py [hunk 1]
```diff
@@ -9,18 +9,32 @@
 # pyre-ignore-all-errors[16]: Undefined attribute of metric states.
 
 import warnings
+from importlib.util import find_spec
 from typing import Iterable, Optional, TypeVar
 
 import torch
 
-from skimage.metrics import structural_similarity
+if find_spec("torchvision") is not None:
```
Comment (bobakfb) [inline torcheval/metrics/image/ssim.py:17]: Shouldn't this be `find_spec("skimage")`?

**Review by bobakfb — CHANGES_REQUESTED** [review]

File: torcheval/metrics/image/ssim.py [hunk 2]
```diff
@@ -9,18 +9,32 @@
...
 from torcheval.metrics.metric import Metric
 
 
 TStructuralSimilarity = TypeVar("TStructuralSimilarity")
 
 
+def _validate_torchvision_available() -> None:
+    if not _SKIMAGE_AVAILABLE:
+        raise RuntimeError(
+                "You must have skimage installed to use SSIM, please install torcheval[image]"
```
Comment (bobakfb) [inline torcheval/metrics/image/ssim.py:34]: extra tab here the linter is angry about

**Comment by bobakfb** [comment]
Hi @fleonce  I've requested two minor changes for this to land, if you can push an update by tomorrow I will take this, otherwise I will just push with these changes on my end.

File: torcheval/metrics/image/ssim.py (hunk 1 above)
Comment (fleonce) [reply torcheval/metrics/image/ssim.py:17]: Absolutely, will fix

File: torcheval/metrics/image/ssim.py (hunk 2 above)
Comment (fleonce) [reply torcheval/metrics/image/ssim.py:34]: :+1:

**Comment by fleonce** [comment]
@bobakfb changes should be up, thanks for reviewing

**Comment by bobakfb** [comment]
sorry looks like another linting issue, can you make sure you run `pre-commit run --all-files` ?

**Comment by fleonce** [comment]
yeah, sure, sorry about that. pre commit runs fine on my machine now. image-requirements.txt needed a new line at the end of the file

**Comment by bobakfb** [comment]
Still breaking tests, but this is a prior failure that is fixed in #207

**Pull request closed without merging**